easybuild/easyconfigs/u/unicore-uftp/unicore-uftp-1.4.2-Java-1.8.eb

```diff
@@ -0,0 +1,28 @@
+easyblock = 'PackedBinary'
+
+name = 'unicore-uftp'
+version = '1.4.2'
+versionsuffix = '-Java-%(javaver)s'
+
+homepage = 'https://sourceforge.net/projects/unicore/'
+description = """UNICORE Java-based client for UFTP"""
+
+toolchain = SYSTEM
+
+source_urls = ['https://downloads.sourceforge.net/project/unicore/Clients/UFTP-Client/%(version)s']
+sources = ['uftp-client-%(version)s-all.zip']
+checksums = ['49670714416077274a0dfdc261e5552bcabbdd5cc4454181a794158491524c54']
+
+# specify dependency on Java/1.8 "wrapper", rather than a specific Java version
+dependencies = [('Java', '1.8', '', True)]
```

Comment:
@DirectorSloan checking the sources here: https://sourceforge.net/projects/unicore/files/Clients/UFTP-Client/1.4.2/

it requires Java 11?

also, please replace `True` with `SYSTEM`

Reply:
@smoors thanks for looking this up. You may did a better job than we as the admin team. We installed it with this dependency and it works this way:

```
login@host 08:09 /tmp/easybuild-build/R $ ml spider unicore-uftp

---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  unicore-uftp:
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
    Description:
      UNICORE Java-based client for UFTP

     Versions:
        unicore-uftp/1.4.2-Java-1.8

---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  For detailed information about a specific "unicore-uftp" module (including how to load the modules) use the module's full name.
  For example:

     $ module spider unicore-uftp/1.4.2-Java-1.8
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

 

login@host 08:09 /tmp/easybuild-build/R $ ml unicore-uftp
login@host 08:10 /tmp/easybuild-build/R $ ml

Currently Loaded Modules:
  1) Java/1.8.0_202   2) unicore-uftp/1.4.2-Java-1.8

 

admbela@frontend2 08:10 /tmp/easybuild-build/R $ uftp
UFTP Client 1.4.2, https://www.unicore.eu
Usage: uftp <command> [OPTIONS] <args>
The following commands are available:
 authenticate         - Authenticates the user and prints out connect info. Optionally specify the initial directory for the FTP session.
 cp                   - Copy file(s). Wildcards '*' are supported.
 get-share            - Download a shared file
 info                 - Gets info about the remote server
 ls                   - Lists a remote directory.
 mkdir                - Creates remote directories.
 put-share            - Upload file(s) to a shared path
 rm                   - Deletes remote files or directories.
 share                - Create, update and delete shares.
 sync                 - Sync a file (the slave) with a master file.
 tunnel               - Forward connections to a local port to a remote socket via UFTPD
Enter 'uftp <command> -h' for help on a particular command.
```

But I will rebuild the tool with Java 11. This is the way it should be I think.

The change from ` True` to `SYSTEM` is no problem.

Reply:
Replaced with Java-11 and tested. Works.